### ISSUE TITLE
munin-async: several plugins do not work

### ISSUE THREAD

**Issue by ingoj** [munin-monitoring/munin#951]
Hi! 

It seems as if several plugins are not working well when used with munin-async: 
```
2018/04/06 21:05:03 [WARNING] Service traffic on tor/ssh://munin-async@tor.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label total
2018/04/06 21:05:03 [WARNING] Service traffic on tor/ssh://munin-async@tor.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv6
2018/04/06 21:05:03 [WARNING] Service traffic on tor/ssh://munin-async@tor.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv4
2018/04/06 21:05:03 [WARNING] Service traffic_percent on tor/ssh://munin-async@tor.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv4_percent
2018/04/06 21:05:03 [WARNING] Service traffic_percent on tor/ssh://munin-async@tor.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label total_percent
2018/04/06 21:05:03 [WARNING] Service traffic_percent on tor/ssh://munin-async@tor.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv6_percent
2018/04/06 21:05:03 [WARNING] Service netstat on tor/ssh://munin-async@tor.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label active
2018/04/06 21:05:04 [WARNING] Service netstat on monitor/ssh://munin-async@monitor.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label active
2018/04/06 21:05:04 [WARNING] Service if_tun2 on vserv/ssh://munin-async@vserv.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label up
2018/04/06 21:05:04 [WARNING] Service if_tun2 on vserv/ssh://munin-async@vserv.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label down
2018/04/06 21:05:04 [WARNING] Service traffic_percent on vserv/ssh://munin-async@vserv.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label total_percent
2018/04/06 21:05:04 [WARNING] Service traffic_percent on vserv/ssh://munin-async@vserv.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv4_percent
2018/04/06 21:05:04 [WARNING] Service traffic_percent on vserv/ssh://munin-async@vserv.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv6_percent
2018/04/06 21:05:04 [WARNING] Service traffic on vserv/ssh://munin-async@vserv.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv6
2018/04/06 21:05:04 [WARNING] Service traffic on vserv/ssh://munin-async@vserv.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv4
2018/04/06 21:05:04 [WARNING] Service traffic on vserv/ssh://munin-async@vserv.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label total
2018/04/06 21:05:04 [WARNING] Service traffic on ax/ssh://munin-async@ax.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv4
2018/04/06 21:05:04 [WARNING] Service traffic on ax/ssh://munin-async@ax.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv6
2018/04/06 21:05:04 [WARNING] Service traffic on ax/ssh://munin-async@ax.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label total
2018/04/06 21:05:05 [WARNING] Service traffic_percent on ax/ssh://munin-async@ax.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv4_percent
2018/04/06 21:05:05 [WARNING] Service traffic_percent on ax/ssh://munin-async@ax.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label total_percent
2018/04/06 21:05:05 [WARNING] Service traffic_percent on ax/ssh://munin-async@ax.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv6_percent
2018/04/06 21:05:05 [WARNING] Service netstat on ax/ssh://munin-async@ax.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label active
2018/04/06 21:05:05 [WARNING] Service netstat on raspi/ssh://munin-async@192.168.0.14 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label active
2018/04/06 21:05:05 [WARNING] Service apc_nis on raspi/ssh://munin-async@192.168.0.14 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label time_left
2018/04/06 21:05:05 [WARNING] Service apc_nis on raspi/ssh://munin-async@192.168.0.14 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label battery_volt
2018/04/06 21:05:05 [WARNING] Service apc_nis on raspi/ssh://munin-async@192.168.0.14 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label load
2018/04/06 21:05:05 [WARNING] Service apc_nis on raspi/ssh://munin-async@192.168.0.14 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label line_volt
2018/04/06 21:05:05 [WARNING] Service apc_nis on raspi/ssh://munin-async@192.168.0.14 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label battery_charge
2018/04/06 21:05:05 [WARNING] Service netstat on raspi2/ssh://munin-async@192.168.0.16 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label active
2018/04/06 21:05:07 [WARNING] Service backuppc_sizes on hahn/ssh://munin-async@hahn.os.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label arrakis_incr
2018/04/06 21:05:07 [WARNING] Service backuppc_sizes on hahn/ssh://munin-async@hahn.os.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label vivaldi_full
2018/04/06 21:05:07 [WARNING] Service backuppc_sizes on hahn/ssh://munin-async@hahn.os.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label arrakis_full
2018/04/06 21:05:07 [WARNING] Service backuppc_sizes on hahn/ssh://munin-async@hahn.os.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label vivaldi_incr
2018/04/06 21:05:07 [WARNING] Service backuppc_ages on hahn/ssh://munin-async@hahn.os.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label arrakis_full
2018/04/06 21:05:07 [WARNING] Service backuppc_ages on hahn/ssh://munin-async@hahn.os.localnet /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label arrakis_incr
2018/04/06 21:05:08 [WARNING] Service traffic on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label total
2018/04/06 21:05:08 [WARNING] Service traffic on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv4
2018/04/06 21:05:08 [WARNING] Service traffic on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv6
2018/04/06 21:05:08 [WARNING] Service xen on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label fido
2018/04/06 21:05:08 [WARNING] Service xen on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label friendos
2018/04/06 21:05:08 [WARNING] Service xen on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label friendica
2018/04/06 21:05:08 [WARNING] Service xen on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label _null_
2018/04/06 21:05:08 [WARNING] Service xen on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label etherpad
2018/04/06 21:05:08 [WARNING] Service xen on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label debian
2018/04/06 21:05:08 [WARNING] Service xen on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label aros
2018/04/06 21:05:08 [WARNING] Service xen on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label john
2018/04/06 21:05:08 [WARNING] Service xen on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label akvorrat
2018/04/06 21:05:09 [WARNING] Service traffic_percent on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv4_percent
2018/04/06 21:05:09 [WARNING] Service traffic_percent on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label total_percent
2018/04/06 21:05:09 [WARNING] Service traffic_percent on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label IPv6_percent
2018/04/06 21:05:09 [WARNING] Service xen_vbd on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label axRD
2018/04/06 21:05:09 [WARNING] Service xen_vbd on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label Domain_0WR
2018/04/06 21:05:09 [WARNING] Service xen_vbd on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label comservRD
2018/04/06 21:05:09 [WARNING] Service xen_vbd on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label comservWR
2018/04/06 21:05:09 [WARNING] Service xen_vbd on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label Domain_0
2018/04/06 21:05:09 [WARNING] Service xen_vbd on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label Domain_0RD
2018/04/06 21:05:09 [WARNING] Service xen_vbd on gate/ssh://munin-async@192.168.254.254 /usr/share/munin/munin-async --spooldir /var/lib/munin-async --spoolfetch:4949 returned no data for label axWR
```

When using the standard way with direct connection to munin-node on port 4949, the plugins are working just fine. Especially squid and traffic plugins are affected. 

If you have tips for better debugging, I'm more than happy to provide more data. Because the plugins are working with direct connection, I assume a bug in munin-async. 

Ingo

**Comment by sumpfralle** [comment]
I could imagine, that 29c2bb8a12e5f46abc4e121bfadc73030d9bd153 fixes these issues.

Could you please verify this?

**Comment by brknkfr** [comment]
> I could imagine, that [29c2bb8](https://github.com/munin-monitoring/munin/commit/29c2bb8a12e5f46abc4e121bfadc73030d9bd153) fixes these issues.
> 
> Could you please verify this?

I had the same problem with the tor plugins (debian stretch with munin from backports). Applying the patch does work, so I can verify this.

**Comment by sumpfralle** [comment]
@brknkfr: thank you for your feedback!

I feel assured that the issue is solved in 2.0.48. Otherwise: please re-open.
